diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/XF86keysym.h b/vcxsrv-code/X11/XF86keysym.h
index c242e42..59ba54d 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/XF86keysym.h
+++ b/vcxsrv-code/X11/XF86keysym.h
@@ -394,6 +394,7 @@
 #define XF86XK_VoiceCommand		_EVDEVK(0x246)		/* v3.16 KEY_VOICECOMMAND */
 #define XF86XK_Assistant		_EVDEVK(0x247)		/* v4.13 KEY_ASSISTANT */
 /* Use: XK_ISO_Next_Group		_EVDEVK(0x248)		   v5.2  KEY_KBD_LAYOUT_NEXT */
+#define XF86XK_EmojiPicker		_EVDEVK(0x249)		/* v5.13 KEY_EMOJI_PICKER */
 #define XF86XK_BrightnessMin		_EVDEVK(0x250)		/* v3.16 KEY_BRIGHTNESS_MIN */
 #define XF86XK_BrightnessMax		_EVDEVK(0x251)		/* v3.16 KEY_BRIGHTNESS_MAX */
 #define XF86XK_KbdInputAssistPrev	_EVDEVK(0x260)		/* v3.18 KEY_KBDINPUTASSIST_PREV */
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xos.h b/vcxsrv-code/X11/Xos.h
index 37d268f..28dfc67 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xos.h
+++ b/vcxsrv-code/X11/Xos.h
@@ -80,7 +80,6 @@ in this Software without prior written authorization from The Open Group.
 #   include <unistd.h>
 #  endif
 #  ifdef WIN32
-#   include <io.h>
 #   include <X11/Xw32defs.h>
 #  else
 #   include <sys/file.h>
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xw32defs.h b/vcxsrv-code/X11/Xw32defs.h
index 683b588..db1eabd 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xw32defs.h
+++ b/vcxsrv-code/X11/Xw32defs.h
@@ -27,7 +27,6 @@ typedef char *caddr_t;
 #  define execve	 _execve
 #  define execvp	 _execvp
 #  define execvpe  _execvpe
-#  define fdopen	  _fdopen
 #  define fileno	  _fileno
 #  define fstat	 _fstat
 #  define getcwd	_getcwd
@@ -72,8 +71,8 @@ typedef char *caddr_t;
 #  define S_IEXEC  _S_IEXEC
 
 #  define	F_OK	0
-#  define	X_OK	1
 #  define	W_OK	2
 #  define	R_OK	4
+#  define	X_OK	R_OK
 # endif /* __GNUC__ */
 #endif
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xwindows.h b/vcxsrv-code/X11/Xwindows.h
index b09356a..1ddf2e3 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xwindows.h
+++ b/vcxsrv-code/X11/Xwindows.h
@@ -1,3 +1,6 @@
+#ifndef __XWINDOWS_H__
+#define __XWINDOWS_H__
+
 /*
 
 Copyright 1996, 1998  The Open Group
@@ -54,7 +57,7 @@ The Open Group.
  *
  */
 #define NOMINMAX
-#define WIN32_LEAN_AND_MEAN // Otherwise this will include winsock.h!
+
 /*
  * mingw-w64 headers define BOOL as a typedef, protecting against macros
  * mingw.org headers define BOOL in terms of WINBOOL
@@ -67,33 +70,27 @@ The Open Group.
 #define INT64 wINT64
 #define LONG64 wLONG64
 #endif
+#ifdef Status
 #undef Status
+#endif
 #define Status wStatus
-#ifdef None // FS.h defines None which is used in winnt.h
+#define ATOM wATOM
+#define BYTE wBYTE
+#define FreeResource wFreeResource
+#ifdef None
 #pragma push_macro("None")
 #define NoneDefined
 #undef None
 #endif
-#ifdef ControlMask // X.h defines ControlMask which is used in processthreadapi.h
+#ifdef ControlMask
 #pragma push_macro("ControlMask")
 #define ControlMaskDefined
 #undef ControlMask
 #endif
-#define ATOM wATOM
-#define BYTE wBYTE
-#define FreeResource wFreeResource
 #include <windows.h>
 #undef NOMINMAX
 #undef Status
 typedef int Status;
-#ifdef NoneDefined
-#pragma pop_macro("None")
-#undef NoneDefined
-#endif
-#ifdef ControlMaskDefined
-#pragma pop_macro("ControlMask")
-#undef ControlMaskDefined
-#endif
 #undef BYTE
 #undef BOOL
 #undef INT32
@@ -102,6 +99,14 @@ typedef int Status;
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
+#ifdef NoneDefined
+#pragma pop_macro("None")
+#undef NoneDefined
+#endif
+#ifdef ControlMaskDefined
+#pragma pop_macro("ControlMask")
+#undef ControlMaskDefined
+#endif
 
 /*
  * Older version of this header used to name the windows API bool type wBOOL,
@@ -130,3 +135,4 @@ typedef int Status;
 # undef _XFree86Server
 #endif
 
+#endif
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xwinsock.h b/vcxsrv-code/X11/Xwinsock.h
index 13360a6..4764930 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/Xwinsock.h
+++ b/vcxsrv-code/X11/Xwinsock.h
@@ -1,3 +1,6 @@
+#ifndef __XWINSOCK_H__
+#define __XWINSOCK_H__
+
 /*
 
 Copyright 1996, 1998  The Open Group
@@ -39,6 +42,10 @@ The Open Group.
 # undef XFree86Server
 #endif
 
+#ifndef FD_SETSIZE
+#define FD_SETSIZE 1024
+#endif
+
 /*
  * mingw-w64 headers define BOOL as a typedef, protecting against macros
  * mingw.org headers define BOOL in terms of WINBOOL
@@ -47,12 +54,13 @@ The Open Group.
 #define _NO_BOOL_TYPEDEF
 #define BOOL WINBOOL
 #define INT32 wINT32
-#ifdef __x86_64__
-#define INT64 wINT64
-#define LONG64 wLONG64
-#endif
+#ifdef Status
 #undef Status
+#endif
 #define Status wStatus
+#define ATOM wATOM
+#define BYTE wBYTE
+#define FreeResource wFreeResource
 #ifdef None
 #pragma push_macro("None")
 #define NoneDefined
@@ -63,31 +71,26 @@ The Open Group.
 #define ControlMaskDefined
 #undef ControlMask
 #endif
-#define ATOM wATOM
-#define BYTE wBYTE
-#define FreeResource wFreeResource
 #include <winsock2.h>
 #include <ws2tcpip.h>
 #undef Status
 typedef int Status;
-#ifdef NoneDefined
-#pragma pop_macro("None")
-#undef NoneDefined
-#endif
-#ifdef ControlMaskDefined
-#pragma pop_macro("ControlMask")
-#undef ControlMaskDefined
-#endif
 #undef BYTE
 #undef BOOL
 #undef INT32
-#undef INT64
-#undef LONG64
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
 #undef RT_FONT
 #undef RT_CURSOR
+#ifdef NoneDefined
+#pragma pop_macro("None")
+#undef NoneDefined
+#endif
+#ifdef ControlMaskDefined
+#pragma pop_macro("ControlMask")
+#undef ControlMaskDefined
+#endif
 
 /*
  * Older version of this header used to name the windows API bool type wBOOL,
@@ -95,8 +98,14 @@ typedef int Status;
  */
 #define wBOOL WINBOOL
 
+#if defined(WIN32) && (!defined(PATH_MAX) || PATH_MAX < 1024)
+# undef PATH_MAX
+# define PATH_MAX 1024
+#endif
+
 #ifdef _XFree86Server
 # define XFree86Server
 # undef _XFree86Server
 #endif
 
+#endif
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/XI2proto.h b/vcxsrv-code/X11/extensions/XI2proto.h
index aaeac7b..c60bd45 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/XI2proto.h
+++ b/vcxsrv-code/X11/extensions/XI2proto.h
@@ -105,6 +105,11 @@
  *                               COMMON STRUCTS                                      *
  *                                                                                   *
  *************************************************************************************/
+#ifdef _MSC_VER
+typedef int int32_t;
+typedef unsigned uint32_t;
+#endif
+
 /** Fixed point 16.16 */
 typedef int32_t FP1616;
 
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/bigreqstr.h b/vcxsrv-code/X11/extensions/bigreqstr.h
index 0a023db..df43f46 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/bigreqstr.h
+++ b/vcxsrv-code/X11/extensions/bigreqstr.h
@@ -1,3 +1,3 @@
-#warning "bigreqstr.h is obsolete and may be removed in the future."
-#warning "include <X11/extensions/bigreqsproto.h> for the protocol defines."
+#pragma message("bigreqstr.h is obsolete and may be removed in the future.")
+#pragma message("include <X11/extensions/bigreqsproto.h> for the protocol defines.")
 #include <X11/extensions/bigreqsproto.h>
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/recordstr.h b/vcxsrv-code/X11/extensions/recordstr.h
index 7f269b7..983af4d 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/recordstr.h
+++ b/vcxsrv-code/X11/extensions/recordstr.h
@@ -1,4 +1,4 @@
-#warning "recordstr.h is obsolete and may be removed in the future."
-#warning "include <X11/extensions/record.h> for the library interfaces."
-#warning "include <X11/extensions/recordproto.h> for the protocol defines."
+#pragma message( "recordstr.h is obsolete and may be removed in the future.")
+#pragma message("include <X11/extensions/record.h> for the library interfaces.")
+#pragma message( "include <X11/extensions/recordproto.h> for the protocol defines.")
 #include <X11/extensions/recordproto.h>
diff --git a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/xcmiscstr.h b/vcxsrv-code/X11/extensions/xcmiscstr.h
index c2b6433..43ab9ff 100644
--- a/xserver-vcpkg/build/ninja/release/_deps/vcpkg-src/buildtrees/xproto/src/f34a46c48d-94a027a0a2.clean/include/X11/extensions/xcmiscstr.h
+++ b/vcxsrv-code/X11/extensions/xcmiscstr.h
@@ -1,3 +1,3 @@
-#warning "xcmiscstr.h is obsolete and may be removed in the future."
-#warning "include <X11/extensions/xcmiscproto.h> for the protocol defines."
+#pragma message( "xcmiscstr.h is obsolete and may be removed in the future."  )
+#pragma message( "include <X11/extensions/xcmiscproto.h> for the protocol defines." )
 #include <X11/extensions/xcmiscproto.h>
