diff --git a/meson.build b/meson.build
index 0aea31260..dad79adc8 100644
--- a/meson.build
+++ b/meson.build
@@ -632,7 +632,7 @@ endif
 
 dep_dxheaders = null_dep
 if with_gallium_d3d12 or with_microsoft_clc
-  dep_dxheaders = dependency('DirectX-Headers', fallback : ['DirectX-Headers', 'dep_dxheaders'],
+  dep_dxheaders = dependency('DirectX-Headers', modules : ['Microsoft::DirectX-Headers'], fallback : ['DirectX-Headers', 'dep_dxheaders'],
     required : with_gallium_d3d12
   )
 endif
