diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/action.c b/vcxsrv-code/xkbcomp/action.c
index 292f2ea..4623c0c 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/action.c
+++ b/vcxsrv-code/xkbcomp/action.c
@@ -436,14 +436,6 @@ HandleSetLatchMods(XkbDescPtr xkb,
     return ReportIllegal(action->type, field);
 }
 
-static LookupEntry lockWhich[] = {
-    {"both", 0},
-    {"lock", XkbSA_LockNoUnlock},
-    {"neither", (XkbSA_LockNoLock | XkbSA_LockNoUnlock)},
-    {"unlock", XkbSA_LockNoLock},
-    {NULL, 0}
-};
-
 static Bool
 HandleLockMods(XkbDescPtr xkb,
                XkbAnyAction * action,
@@ -451,19 +443,12 @@ HandleLockMods(XkbDescPtr xkb,
 {
     XkbModAction *act;
     unsigned t1, t2;
-    ExprResult rtrn;
 
     act = (XkbModAction *) action;
-    if ((array_ndx != NULL) && (field == F_Modifiers || field == F_Affect))
+    if ((array_ndx != NULL) && (field == F_Modifiers))
         return ReportActionNotArray(action->type, field);
     switch (field)
     {
-    case F_Affect:
-        if (!ExprResolveEnum(value, &rtrn, lockWhich))
-            return ReportMismatch(action->type, field, "lock or unlock");
-        act->flags &= ~(XkbSA_LockNoLock | XkbSA_LockNoUnlock);
-        act->flags |= rtrn.uval;
-        return True;
     case F_Modifiers:
         t1 = act->flags;
         if (CheckModifierField(xkb, action->type, value, &t1, &t2))
@@ -642,7 +627,6 @@ HandleMovePtr(XkbDescPtr xkb,
             act->flags &= ~XkbSA_NoAcceleration;
         else
             act->flags |= XkbSA_NoAcceleration;
-        return True;
     }
     return ReportIllegal(action->type, field);
 }
@@ -657,6 +641,14 @@ static LookupEntry btnNames[] = {
     {NULL, 0}
 };
 
+static LookupEntry lockWhich[] = {
+    {"both", 0},
+    {"lock", XkbSA_LockNoUnlock},
+    {"neither", (XkbSA_LockNoLock | XkbSA_LockNoUnlock)},
+    {"unlock", XkbSA_LockNoLock},
+    {NULL, 0}
+};
+
 static Bool
 HandlePtrBtn(XkbDescPtr xkb,
              XkbAnyAction * action,
@@ -690,7 +682,7 @@ HandlePtrBtn(XkbDescPtr xkb,
         if (!ExprResolveEnum(value, &rtrn, lockWhich))
             return ReportMismatch(action->type, field, "lock or unlock");
         act->flags &= ~(XkbSA_LockNoLock | XkbSA_LockNoUnlock);
-        act->flags |= rtrn.uval;
+        act->flags |= rtrn.ival;
         return True;
     }
     else if (field == F_Count)
@@ -787,12 +779,8 @@ static LookupEntry isoNames[] = {
     {"pointer", XkbSA_ISONoAffectPtr},
     {"ctrls", XkbSA_ISONoAffectCtrls},
     {"controls", XkbSA_ISONoAffectCtrls},
-    {"all", XkbSA_ISOAffectMask},
+    {"all", ~((unsigned) 0)},
     {"none", 0},
-    {"both", 0},
-    {"lock", XkbSA_LockNoUnlock},
-    {"neither", (XkbSA_LockNoLock | XkbSA_LockNoUnlock)},
-    {"unlock", XkbSA_LockNoLock},
     {NULL, 0},
 };
 
@@ -816,8 +804,8 @@ HandleISOLock(XkbDescPtr xkb,
         if (CheckModifierField(xkb, action->type, value, &flags, &mods))
         {
             act->flags = flags & (~XkbSA_ISODfltIsGroup);
-            act->real_mods = act->mask = (mods & 0xff);
-            mods = (mods >> 8) & 0xffff;
+            act->real_mods = mods & 0xff;
+            mods = (mods >> 8) & 0xff;
             XkbSetModActionVMods(act, mods);
             return True;
         }
@@ -839,8 +827,6 @@ HandleISOLock(XkbDescPtr xkb,
         if (!ExprResolveMask(value, &rtrn, SimpleLookup, (XPointer) isoNames))
             return ReportMismatch(action->type, field, "keyboard component");
         act->affect = (~rtrn.uval) & XkbSA_ISOAffectMask;
-        act->flags &= ~(XkbSA_LockNoLock | XkbSA_LockNoUnlock);
-        act->flags |= rtrn.uval & (XkbSA_LockNoLock | XkbSA_LockNoUnlock);
         return True;
     }
     return ReportIllegal(action->type, field);
@@ -957,15 +943,6 @@ HandleSetLockControls(XkbDescPtr xkb,
         XkbActionSetCtrls(act, rtrn.uval);
         return True;
     }
-    else if (field == F_Affect && action->type == XkbSA_LockControls) {
-        if (array_ndx != NULL)
-            return ReportActionNotArray(action->type, field);
-        if (!ExprResolveEnum(value, &rtrn, lockWhich))
-            return ReportMismatch(action->type, field, "lock or unlock");
-        act->flags &= ~(XkbSA_LockNoLock | XkbSA_LockNoUnlock);
-        act->flags |= rtrn.uval;
-        return True;
-    }
     return ReportIllegal(action->type, field);
 }
 
@@ -1145,7 +1122,7 @@ HandleDeviceBtn(XkbDescPtr xkb,
         if (!ExprResolveEnum(value, &rtrn, lockWhich))
             return ReportMismatch(action->type, field, "lock or unlock");
         act->flags &= ~(XkbSA_LockNoLock | XkbSA_LockNoUnlock);
-        act->flags |= rtrn.uval;
+        act->flags |= rtrn.ival;
         return True;
     }
     else if (field == F_Count)
@@ -1312,7 +1289,7 @@ ApplyActionFactoryDefaults(XkbAction * action)
     }
     else if (action->type == XkbSA_ISOLock)
     {
-        action->iso.real_mods = action->iso.mask = LockMask;
+        action->iso.real_mods = LockMask;
     }
     return;
 }
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/autogen.sh b/vcxsrv-code/xkbcomp/autogen.sh
index 2c5bc91..fc34bd5 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/autogen.sh
+++ b/vcxsrv-code/xkbcomp/autogen.sh
@@ -1,17 +1,14 @@
 #! /bin/sh
 
-srcdir=`dirname "$0"`
+srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
 
 ORIGDIR=`pwd`
-cd "$srcdir"
+cd $srcdir
 
 autoreconf -v --install || exit 1
-cd "$ORIGDIR" || exit $?
-
-git config --local --get format.subjectPrefix >/dev/null 2>&1 ||
-    git config --local format.subjectPrefix "PATCH app/xkbcomp"
+cd $ORIGDIR || exit $?
 
 if test -z "$NOCONFIGURE"; then
-    exec "$srcdir"/configure "$@"
+    $srcdir/configure "$@"
 fi
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/compat.c b/vcxsrv-code/xkbcomp/compat.c
index 82fd4e6..f9402f1 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/compat.c
+++ b/vcxsrv-code/xkbcomp/compat.c
@@ -114,7 +114,11 @@ InitCompatInfo(CompatInfo * info, XkbDescPtr xkb)
     info->dflt.defs.fileID = info->fileID;
     info->dflt.defs.defined = 0;
     info->dflt.defs.merge = MergeOverride;
+    info->dflt.defs.next = NULL;
     info->dflt.interp.flags = 0;
+    info->dflt.interp.match = 0;
+    info->dflt.interp.mods = 0;
+    info->dflt.interp.sym = 0;
     info->dflt.interp.virtual_mod = XkbNoModifier;
     info->dflt.interp.act.type = XkbSA_NoAction;
     for (i = 0; i < XkbAnyActionDataSize; i++)
@@ -655,14 +659,13 @@ HandleInterpDef(InterpDef * def, XkbDescPtr xkb, unsigned merge,
 
     if (!ResolveStateAndPredicate(def->match, &pred, &mods, info))
     {
-        ERROR("Couldn't determine matching modifiers\n");
+        if (warningLevel > 0)
+          WARN1("Couldn't determine matching modifiers\n");
         ACTION("Symbol interpretation ignored\n");
         return True;
     }
     if (def->ignore)
     {
-        ERROR("Couldn't lookup keysym\n");
-        ACTION("Symbol interpretation ignored\n");
         return True;
     }
 
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/configure.ac b/vcxsrv-code/xkbcomp/configure.ac
index 2aa0d1d..0c8063e 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/configure.ac
+++ b/vcxsrv-code/xkbcomp/configure.ac
@@ -22,8 +22,8 @@ dnl Process this file with autoconf to create configure.
 
 # Initialize Autoconf
 AC_PREREQ([2.60])
-AC_INIT([xkbcomp], [1.4.3],
-        [https://gitlab.freedesktop.org/xorg/app/xkbcomp/issues], [xkbcomp])
+AC_INIT([xkbcomp], [1.3.1],
+        [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], [xkbcomp])
 AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_HEADERS([config.h])
 
@@ -47,8 +47,11 @@ fi
 
 AC_CHECK_FUNCS([strdup strcasecmp])
 
+REQUIRED_MODULES="x11 xkbfile xproto >= 7.0.17"
+
 # Checks for pkg-config packages
-PKG_CHECK_MODULES(XKBCOMP, [x11 xkbfile x11-xcb xau xdmcp xproto >= 7.0.17])
+PKG_CHECK_MODULES(XKBCOMP, [$REQUIRED_MODULES])
+AC_SUBST(REQUIRED_MODULES)
 
 AC_ARG_WITH([xkb_config_root],
     [AS_HELP_STRING([--with-xkb-config-root=<paths>],
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/keycodes.c b/vcxsrv-code/xkbcomp/keycodes.c
index 425ba10..57c8bc8 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/keycodes.c
+++ b/vcxsrv-code/xkbcomp/keycodes.c
@@ -31,8 +31,6 @@
 #include "misc.h"
 #include "alias.h"
 
-static Bool high_keycode_warned;
-
 char *
 longText(unsigned long val, unsigned format)
 {
@@ -299,6 +297,8 @@ InitKeyNamesInfo(KeyNamesInfo * info)
     info->name = NULL;
     info->leds = NULL;
     info->aliases = NULL;
+    info->fileID=-1;
+    info->merge=0;
     ClearKeyNamesInfo(info);
     info->errorCount = 0;
     return;
@@ -332,15 +332,10 @@ AddKeyName(KeyNamesInfo * info,
 
     if ((kc < info->effectiveMin) || (kc > info->effectiveMax))
     {
-        if (!high_keycode_warned && warningLevel > 1)
-        {
-            WARN2("Unsupported high keycode %d for name <%s> ignored\n",
-                  kc, name);
-            ACTION2("X11 cannot support keycodes above 255.\n");
-            ACTION2("This warning only shows for the first high keycode.\n");
-            high_keycode_warned = True;
-        }
-        return True;
+        ERROR2("Illegal keycode %d for name <%s>\n", kc, name);
+        ACTION2("Must be in the range %d-%d inclusive\n",
+                info->effectiveMin, info->effectiveMax);
+        return False;
     }
     if (kc < info->computedMin)
         info->computedMin = kc;
@@ -596,15 +591,10 @@ HandleKeycodeDef(KeycodeDef * stmt, unsigned merge, KeyNamesInfo * info)
     code = result.ival;
     if ((code < info->effectiveMin) || (code > info->effectiveMax))
     {
-        if (!high_keycode_warned && warningLevel > 1)
-        {
-            WARN2("Unsupported high keycode %d for name <%s> ignored\n",
-                  code, stmt->name);
-            ACTION2("X11 cannot support keycodes above 255.\n");
-            ACTION2("This warning only shows for the first high keycode.\n");
-            high_keycode_warned = True;
-        }
-        return 1;
+        ERROR2("Illegal keycode %d for name <%s>\n", code, stmt->name);
+        ACTION2("Must be in the range %d-%d inclusive\n",
+                info->effectiveMin, info->effectiveMax);
+        return 0;
     }
     if (stmt->merge != MergeDefault)
     {
@@ -663,7 +653,7 @@ HandleKeyNameVar(VarDef * stmt, KeyNamesInfo * info)
         ACTION1("Assignment to field %s ignored\n", field.str);
         return 0;
     }
-    if ((tmp.ival < XkbMinLegalKeyCode))
+    if ((tmp.ival < XkbMinLegalKeyCode) || (tmp.ival > XkbMaxLegalKeyCode))
     {
         ERROR3
             ("Illegal keycode %d (must be in the range %d-%d inclusive)\n",
@@ -671,14 +661,6 @@ HandleKeyNameVar(VarDef * stmt, KeyNamesInfo * info)
         ACTION1("Value of \"%s\" not changed\n", field.str);
         return 0;
     }
-    if ((tmp.ival > XkbMaxLegalKeyCode))
-    {
-        WARN2("Unsupported maximum keycode %d, clipping.\n", tmp.ival);
-        ACTION2("X11 cannot support keycodes above 255.\n");
-        info->explicitMax = XkbMaxLegalKeyCode;
-        info->effectiveMax = XkbMaxLegalKeyCode;
-        return 1;
-    }
     if (which == MIN_KEYCODE_DEF)
     {
         if ((info->explicitMax > 0) && (info->explicitMax < tmp.ival))
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/listing.c b/vcxsrv-code/xkbcomp/listing.c
index ab61a58..aaab61a 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/listing.c
+++ b/vcxsrv-code/xkbcomp/listing.c
@@ -74,10 +74,8 @@ SOFTWARE.
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#ifdef _MSC_VER
-#define S_ISDIR(mode)  (((mode) & S_IFMT) == S_IFDIR)
-#endif
 #include <X11/keysym.h>
+#include <X11/Xwindows.h>
 
 #if defined(sgi)
 #include <malloc.h>
@@ -127,6 +125,14 @@ SOFTWARE.
 #include "tokens.h"
 #include <X11/extensions/XKBgeom.h>
 
+#ifndef S_ISDIR
+# if defined(_S_IFMT) && defined(_S_IFDIR)
+#  define S_ISDIR(a)	(((a) & _S_IFMT) == _S_IFDIR)
+# else
+#  define S_ISDIR(a)	(((a) & S_IFMT) == S_IFDIR)
+# endif
+#endif
+
 #define	lowbit(x)	((x) & (-(x)))
 
 unsigned int listingDebug;
@@ -293,6 +299,7 @@ AddDirectory(char *head, char *ptrn, char *rest, char *map)
 #ifdef WIN32
     if ((dirh = FindFirstFile("*.*", &file)) == INVALID_HANDLE_VALUE)
         return 0;
+    nMatch = 0;
 #else
     if ((dirp = opendir((head ? head : "."))) == NULL)
         return 0;
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/misc.h b/vcxsrv-code/xkbcomp/misc.h
index 4fa4b6d..015ad5c 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/misc.h
+++ b/vcxsrv-code/xkbcomp/misc.h
@@ -30,7 +30,7 @@
 typedef struct _CommonInfo
 {
     unsigned short defined;
-    unsigned char fileID;
+    int fileID;
     unsigned char merge;
     struct _CommonInfo *next;
 } CommonInfo;
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/parseutils.c b/vcxsrv-code/xkbcomp/parseutils.c
index 5f9f6a4..e98adf1 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/parseutils.c
+++ b/vcxsrv-code/xkbcomp/parseutils.c
@@ -229,7 +229,12 @@ InterpCreate(const char *sym_str, ExprDef * match)
         def->common.stmtType = StmtInterpDef;
         def->common.next = NULL;
         if (LookupKeysym(sym_str, &def->sym) == 0)
+        {
             def->ignore = True;
+            if (warningLevel > 0)
+                WARN1("Couldn't lookup keysym %s\n", sym_str);
+            ACTION("Symbol interpretation ignored\n");
+        }
         else
             def->ignore = False;
         def->match = match;
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/utils.c b/vcxsrv-code/xkbcomp/utils.c
index 541c33e..53dc994 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/utils.c
+++ b/vcxsrv-code/xkbcomp/utils.c
@@ -352,7 +352,7 @@ uStrCaseCmp(const char *str1, const char *str2)
 }
 
 int
-uStrCasePrefix(const char *my_prefix, char *str)
+uStrCasePrefix(const char *my_prefix, const char *str)
 {
     char c1;
     char c2;
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/utils.h b/vcxsrv-code/xkbcomp/utils.h
index 81f5893..e51d1a0 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/utils.h
+++ b/vcxsrv-code/xkbcomp/utils.h
@@ -141,6 +141,10 @@ uInformation(const char * /* s */ , ...
      extern void uWarning(const char * /* s  */ , ...
     ) _X_ATTRIBUTE_PRINTF(1, 2);
 
+#ifdef ERROR
+#undef ERROR
+#endif
+
 #define ERROR6			uError
 #define ERROR5			uError
 #define ERROR4			uError
@@ -206,7 +210,7 @@ uInformation(const char * /* s */ , ...
                             const char *        /* s2 */
     );
      extern int uStrCasePrefix(const char * /* p */ ,
-                               char *   /* str */
+                               const char * /* str */
     );
 #endif
 #ifdef HAVE_STRDUP
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.c b/vcxsrv-code/xkbcomp/xkbcomp.c
index 3065f87..bafef43 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.c
+++ b/vcxsrv-code/xkbcomp/xkbcomp.c
@@ -31,14 +31,7 @@
 /* for symlink attack security fix -- Branden Robinson */
 #include <sys/stat.h>
 #include <sys/types.h>
-#ifdef HAVE_UNISTD_H
 #include <unistd.h>
-#endif
-#if defined(_MSC_VER)
-#include <direct.h>
-#define S_IRUSR S_IREAD
-#define S_IWUSR S_IWRITE
-#endif
 /* end BR */
 
 #if defined(sgi)
@@ -232,7 +225,7 @@ parseArgs(int argc, char *argv[])
     for (i = 1; i < argc; i++)
     {
         int itmp;
-        if ((argv[i][0] != '-') || (strcmp(argv[i], "-") == 0))
+        if ((argv[i][0] != '-') || (uStringEqual(argv[i], "-")))
         {
             if (!xkblist)
             {
@@ -556,8 +549,8 @@ parseArgs(int argc, char *argv[])
                 {
                     WARN1("Changing root directory to \"%s\"\n", rootDir);
                 }
-                if ((chdir(rootDir) < 0) && (warningLevel > 0))
-                {
+		XkbAddDirectoryToPath(rootDir);
+		if (!XkbAddDirectoryToPath(rootDir) && (warningLevel>0)) {
                     WARN1("Couldn't change directory to \"%s\"\n", rootDir);
                     ACTION("Root directory (-R) option ignored\n");
                     rootDir = NULL;
@@ -658,7 +651,7 @@ parseArgs(int argc, char *argv[])
         ERROR("No input file specified\n");
         return False;
     }
-    else if (strcmp(inputFile, "-") == 0)
+    else if (uStringEqual(inputFile, "-"))
     {
         inputFormat = INPUT_XKB;
     }
@@ -762,7 +755,7 @@ parseArgs(int argc, char *argv[])
             return False;
         }
     }
-    else if ((!outputFile) && (inputFile) && (strcmp(inputFile, "-") == 0))
+    else if ((!outputFile) && (inputFile) && uStringEqual(inputFile, "-"))
     {
         int len = strlen("stdin") + strlen(fileTypeExt[outputFormat]) + 2;
         outputFile = uTypedCalloc(len, char);
@@ -937,7 +930,7 @@ main(int argc, char *argv[])
     }
     if (inputFile != NULL)
     {
-        if (strcmp(inputFile, "-") == 0)
+        if (uStringEqual(inputFile, "-"))
         {
             file = stdin;
             inputFile = "stdin";
@@ -1139,7 +1132,7 @@ main(int argc, char *argv[])
         }
         if (outputFile != NULL)
         {
-            if (strcmp(outputFile, "-") == 0)
+            if (uStringEqual(outputFile, "-"))
                 outputFile = "stdout";
             else
             {
@@ -1167,10 +1160,14 @@ main(int argc, char *argv[])
                     break;
                 }
 #endif
+#ifdef _MSC_VER
+		outputFileFd= open(outputFile, O_WRONLY|O_CREAT|O_EXCL|binMode,_S_IREAD | _S_IWRITE);
+#else
                 outputFileFd =
                     open(outputFile, O_WRONLY | O_CREAT | O_EXCL,
                          S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH
                          | S_IWOTH | binMode);
+#endif
                 if (outputFileFd < 0)
                 {
                     ERROR1
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.h b/vcxsrv-code/xkbcomp/xkbcomp.h
index fb40061..c224a2e 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.h
+++ b/vcxsrv-code/xkbcomp/xkbcomp.h
@@ -31,6 +31,7 @@
 #define	DEBUG_VAR debugFlags
 #endif
 
+#include <unistd.h>
 #include <X11/Xlib.h>
 #include <X11/XKBlib.h>
 
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.pc.in b/vcxsrv-code/xkbcomp/xkbcomp.pc.in
index ef5bf6d..b8d6023 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbcomp.pc.in
+++ b/vcxsrv-code/xkbcomp/xkbcomp.pc.in
@@ -1,6 +1,4 @@
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
 datarootdir=@datarootdir@
 datadir=@datadir@
 xkbconfigdir=@XKBCONFIGROOT@
@@ -8,3 +6,4 @@ xkbconfigdir=@XKBCONFIGROOT@
 Name: xkbcomp
 Description: XKB keymap compiler
 Version: @PACKAGE_VERSION@
+Requires.private: @REQUIRED_MODULES@
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbparse.y b/vcxsrv-code/xkbcomp/xkbparse.y
index d816bee..87dd07f 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbparse.y
+++ b/vcxsrv-code/xkbcomp/xkbparse.y
@@ -92,6 +92,10 @@
 #ifdef DEBUG
 #define	YYDEBUG 1
 #endif
+
+#define YYMALLOC malloc
+#define YYFREE free
+
 #define	DEBUG_VAR parseDebug
 #include "parseutils.h"
 #include <X11/keysym.h>
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbpath.c b/vcxsrv-code/xkbcomp/xkbpath.c
index e7912bf..59f5645 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbpath.c
+++ b/vcxsrv-code/xkbcomp/xkbpath.c
@@ -30,9 +30,14 @@
 #define	DEBUG_VAR debugFlags
 #include "utils.h"
 #include <stdlib.h>
+#include <unistd.h>
 #include <X11/extensions/XKM.h>
 #include "xkbpath.h"
 
+#ifndef DFLT_XKB_CONFIG_ROOT
+#define DFLT_XKB_CONFIG_ROOT	"xkbdata"
+#endif
+
 #ifndef PATH_MAX
 #define	PATH_MAX 1024
 #endif
diff --git a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbscan.c b/vcxsrv-code/xkbcomp/xkbscan.c
index 24c7f4d..9a265d3 100644
--- a/xserver-vcpkg/vcpkg/buildtrees/xkbcomp/src/92d23ef874-c15b381187.clean/xkbscan.c
+++ b/vcxsrv-code/xkbcomp/xkbscan.c
@@ -32,7 +32,7 @@
 #include <X11/XKBlib.h>
 
 #include "tokens.h"
-#define	DEBUG_VAR	scanDebug
+
 #include "utils.h"
 #include "parseutils.h"
 
@@ -55,7 +55,7 @@ static int readBufPos = 0;
 static int readBufLen = 0;
 
 #ifdef DEBUG
-extern int debugFlags;
+extern unsigned int debugFlags;
 
 static char *
 tokText(int tok)
@@ -614,9 +614,9 @@ yyGetIdent(int first)
 static int
 yyGetNumber(int ch)
 {
-    const int nMaxBuffSize = 1024;
+    #define nMaxBuffSize 1024
     int isFloat = 0;
-    char buf[1024];
+    char buf[nMaxBuffSize];
     int nInBuf = 0;
 
     buf[0] = ch;
