--- ./hw/xwin/save_winmsg.c	2016-03-10 13:33:02.023919268 +0000
+++ ./hw/xwin/winmsg.c	2016-03-10 13:26:21.055618778 +0000
@@ -129,8 +129,8 @@
             getenv("WIN_DEBUG_WM_USER")) {
             winDebug("%s - Message WM_USER + %d\n", function,
                      message - WM_USER);
-            winDebug("\thwnd 0x%p wParam 0x%x lParam 0x%x\n", hwnd, (int)wParam,
-                     (int)lParam);
+            winDebug("\thwnd 0x%p wParam 0x%x lParam 0x%x timestamp %d\n", hwnd, (int)wParam,
+                     (int)lParam, GetMessageTime());
         }
     }
     else if (message < MESSAGE_NAMES_LEN && MESSAGE_NAMES[message]) {
@@ -141,8 +141,8 @@
         buffer[63] = 0;
         if (force || getenv("WIN_DEBUG_MESSAGES") || getenv(buffer)) {
             winDebug("%s - Message %s\n", function, MESSAGE_NAMES[message]);
-            winDebug("\thwnd 0x%p wParam 0x%x lParam 0x%x\n", hwnd, (int)wParam,
-                     (int)lParam);
+            winDebug("\thwnd 0x%p wParam 0x%x lParam 0x%x timestamp %d\n", hwnd, (int)wParam,
+                     (int)lParam, GetMessageTime());
         }
     }
 }
