--- ./glx/save_indirect_texture_compression.c	2017-06-21 09:47:22.055060486 +0100
+++ ./glx/indirect_texture_compression.c	2017-06-21 09:49:45.172813244 +0100
@@ -39,7 +39,7 @@
 #include "dispatch.h"
 
 int
-__glXDisp_GetCompressedTexImageARB(struct __GLXclientStateRec *cl, GLbyte * pc)
+__glXDisp_GetCompressedTexImage(struct __GLXclientStateRec *cl, GLbyte * pc)
 {
     xGLXSingleReq *const req = (xGLXSingleReq *) pc;
     int error;
@@ -64,7 +64,7 @@
         if (compsize != 0) {
             __GLX_GET_ANSWER_BUFFER(answer, cl, compsize, 1);
             __glXClearErrorOccured();
-            CALL_GetCompressedTexImageARB(GET_DISPATCH(),
+            CALL_GetCompressedTexImage(GET_DISPATCH(),
                                           (target, level, answer));
         }
 
@@ -86,7 +86,7 @@
 }
 
 int
-__glXDispSwap_GetCompressedTexImageARB(struct __GLXclientStateRec *cl,
+__glXDispSwap_GetCompressedTexImage(struct __GLXclientStateRec *cl,
                                        GLbyte * pc)
 {
     xGLXSingleReq *const req = (xGLXSingleReq *) pc;
@@ -113,7 +113,7 @@
         if (compsize != 0) {
             __GLX_GET_ANSWER_BUFFER(answer, cl, compsize, 1);
             __glXClearErrorOccured();
-            CALL_GetCompressedTexImageARB(GET_DISPATCH(),
+            CALL_GetCompressedTexImage(GET_DISPATCH(),
                                           (target, level, answer));
         }
 
