--- ./glx/save_render2.c	2016-05-03 12:57:08.726315122 +0100
+++ ./glx/render2.c	2016-05-03 12:58:40.059163393 +0100
@@ -241,12 +241,12 @@
             break;
         case GL_SECONDARY_COLOR_ARRAY:
             CALL_EnableClientState(GET_DISPATCH(), (GL_SECONDARY_COLOR_ARRAY));
-            CALL_SecondaryColorPointerEXT(GET_DISPATCH(),
+            CALL_SecondaryColorPointer(GET_DISPATCH(),
                                           (numVals, datatype, stride, pc));
             break;
         case GL_FOG_COORD_ARRAY:
             CALL_EnableClientState(GET_DISPATCH(), (GL_FOG_COORD_ARRAY));
-            CALL_FogCoordPointerEXT(GET_DISPATCH(), (datatype, stride, pc));
+            CALL_FogCoordPointer(GET_DISPATCH(), (datatype, stride, pc));
             break;
         default:
             break;
