--- ./hw/xwin/save_winSetAppUserModelID.c	2014-10-27 23:08:17.015874013 +0000
+++ ./hw/xwin/winSetAppUserModelID.c	2014-10-27 22:57:19.213569761 +0000
@@ -25,24 +25,40 @@
 #include <xwin-config.h>
 #endif
 
-#include <X11/Xlib.h>
-#include <X11/Xproto.h>
-#include <X11/Xwindows.h>
-#include <pthread.h>
-#include "winwindow.h"
-#include "os.h"
-#include "winmsg.h"
-
+#include <windows.h>
+#include <ctype.h>
 #include <shlwapi.h>
-
-#define INITGUID
-#include "initguid.h"
 #include "propertystore.h"
-#undef INITGUID
 
 static HMODULE g_hmodShell32Dll = NULL;
 static SHGETPROPERTYSTOREFORWINDOWPROC g_pSHGetPropertyStoreForWindow = NULL;
 
+extern void ErrorF(const char *f, ...);
+extern void winDebug(const char *format, ...);
+
+extern void winPropertyStoreInit(void);
+
+extern void winPropertyStoreDestroy(void);
+
+extern void winSetAppUserModelID(HWND hWnd, char *AppID);
+
+static void
+RemoveWhitespaceAndTruncate(HWND hWnd, char *AppID)
+{
+    char *p;
+    size_t len = strlen(AppID);
+
+    for (p = AppID; *p; p++, len--) {
+        while (isspace(*p))
+            memmove(p, p + 1, len--);
+    }
+    if (strlen(AppID) > 128) {
+        AppID[128] = '\0';
+        ErrorF("RemoveWhitespaceAndTruncate - AppUserModelID truncated for window 0x%p.\n",
+               hWnd);
+    }
+}
+
 void
 winPropertyStoreInit(void)
 {
@@ -80,7 +96,7 @@
 }
 
 void
-winSetAppUserModelID(HWND hWnd, const char *AppID)
+winSetAppUserModelID(HWND hWnd, char *AppID)
 {
     PROPVARIANT pv;
     IPropertyStore *pps = NULL;
@@ -90,13 +106,13 @@
         return;
     }
 
-    winDebug("winSetAppUserMOdelID - hwnd 0x%p appid '%s'\n", hWnd, AppID);
-
     hr = g_pSHGetPropertyStoreForWindow(hWnd, &IID_IPropertyStore,
                                         (void **) &pps);
     if (SUCCEEDED(hr) && pps) {
-        memset(&pv, 0, sizeof(PROPVARIANT));
+        PropVariantInit(&pv);
         if (AppID) {
+            RemoveWhitespaceAndTruncate(hWnd, AppID);
+            winDebug("winSetAppUserMOdelID - hwnd 0x%p appid '%s'\n", hWnd, AppID);
             pv.vt = VT_LPWSTR;
             hr = SHStrDupA(AppID, &pv.pwszVal);
         }
